src/mlpack/methods/neat/selection_strategies/roulette_selection.hpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file roulette_selection.hpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * Definition of the Roulette Selection class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_NEAT_SELECTION_ROULETTE_HPP
+#define MLPACK_METHODS_NEAT_SELECTION_ROULETTE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+class RouletteSelection
```

Comment:
Can you write a test for each selection strategy.

Reply:
So would that basically be an XOR test with all the different selection strategies?

Reply:
I thought we could write some specific tests for the strategies, pass in some defined fitness values and check that the output it correct, the XOR test would be an addition.